src/vm/gchelpers.cpp

```diff
@@ -350,13 +350,28 @@ OBJECTREF AllocateArrayEx(TypeHandle arrayType, INT32 *pArgs, DWORD dwNumArgs, B
                 lowerBound = pArgs[i];
                 i++;
             }
+
             int length = pArgs[i];
             if (length < 0)
                 COMPlusThrow(kOverflowException);
+
             if ((SIZE_T)length > MaxArrayLength(componentSize))
+            {
+                // This will cause us to throw below if we don't throw anything else before then.
                 maxArrayDimensionLengthOverflow = true;
-            if ((length > 0) && (lowerBound + (length - 1) < lowerBound))
```

Comment:
Yes, I considered that approach as well and am willing to change it. Is it any simpler though? The current way seems to be very explicit, clear, and simple to me.